### PR TITLE
CI: Use Pkg.jl for installing GAP.jl

### PR DESCRIPTION
**Pull request by lgoettgens**
Follow-up to https://github.com/gap-system/gap/pull/5733.

~~This is currently WIP.~~

I am trying to adapt this as suggested in https://github.com/gap-system/gap/pull/5733#issuecomment-2145529488.

## Text for release notes
none

### CODE REVIEW

**Comment by lgoettgens** [comment]
Somehow the new CI job doesn't run on this PR at all, so I cannot test it. @fingolfin do you have any idea why?

File: .github/workflows/gapjl.yml [hunk 1]
```diff
@@ -65,7 +74,7 @@ jobs:
           make -j`nproc`
       - name: "Override bundled GAP"
         run: |
-          julia --proj=override etc/setup_override_dir.jl /tmp/GAPROOT /tmp/gap_jll_override
+          julia --proj=override $GAPJLPATH/etc/setup_override_dir.jl /tmp/GAPROOT /tmp/gap_jll_override
       - name: "Run tests"
         run: |
-           julia --proj=override etc/run_with_override.jl /tmp/gap_jll_override --depwarn=error -e "using Pkg; Pkg.test(\"GAP\")"
+           julia --proj=override $GAPJLPATHetc/run_with_override.jl /tmp/gap_jll_override --depwarn=error -e "using Pkg; Pkg.test(\"GAP\")"
```
Comment (fingolfin) [inline .github/workflows/gapjl.yml:80]:
```suggestion
           julia --proj=override $GAPJLPATH/etc/run_with_override.jl /tmp/gap_jll_override --depwarn=error -e "using Pkg; Pkg.test(\"GAP\")"
```

File: .github/workflows/gapjl.yml [hunk 2]
```diff
@@ -40,10 +41,18 @@ jobs:
...
+          if ${{ matrix.gapjl-version }} == "latest-release"
+            Pkg.add("GAP")
+          else
+            Pkg.add(;name="GAP", rev="${{ matrix.gapjl-version }}")
+          end
+          path = dirname(dirname(Base.find_package("GAP")))
+          open(ENV["GITHUB_ENV"], "a") do io
+              println(io, "GAPJLPATH=$path")
+          end
         uses: actions/checkout@v4
```
Comment (fingolfin) [inline .github/workflows/gapjl.yml:55]:
```suggestion
```
I think this was a syntax error

File: .github/workflows/gapjl.yml (hunk 1 above)
Comment (lgoettgens) [reply .github/workflows/gapjl.yml]: good catch

File: .github/workflows/gapjl.yml (hunk 2 above)
Comment (lgoettgens) [reply .github/workflows/gapjl.yml]: Yeah, github needed a minute longer than expected to show it in https://github.com/gap-system/gap/actions/runs/9370508249

**Comment by lgoettgens** [comment]
Testing against the GAP.jl master still works. Testing with the latest release will keep failing until the change in https://github.com/oscar-system/GAP.jl/pull/995#discussion_r1613802107 is released.

**Comment by lgoettgens** [comment]
WIth the new Gap.jl 0.11.1 release, all jobs now succeed.

**Review by fingolfin — APPROVED** [review]

**Pull request merged**